Mention restriction on usage of UnmangedType.CustomMarshaler

dlech opened the pull request:
Copy sentence from UnmangedType.LPStruct to UnmangedType.CustomMarshaler since the same restriction applies.

See dotnet/coreclr#12029

dlech committed:
Mention restriction on usage of UnmangedType.CustomMarshaler

Copy sentence from UnmangedType.LPStruct to UnmangedType.CustomMarshaler since the same restriction applies.

See dotnet/coreclr#12029

dnfclas commented:
This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

yizhang82 commented on xml/System.Runtime.InteropServices/UnmanagedType.xml:
```diff
@@ -291,7 +291,7 @@
       </ReturnValue>
       <Docs>
-        <summary>Specifies the custom marshaler class when used with the <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalType" /> or <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalTypeRef" /> field. The <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalCookie" /> field can be used to pass additional information to the custom marshal ...
+        <summary>Specifies the custom marshaler class when used with the <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalType" /> or <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalTypeRef" /> field. The <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalCookie" /> field can be used to pass additional information to the custom marshal ...
```
This also works on COM interop. So this statement is not accurate.
How about: 
This member is valid for parameters and return values only. It cannot be used on struct fields.

mairaw commented:
@dlech let us know if you'll make the changes suggested by @yizhang82. Otherwise, we can make them for you in this PR. Thanks!

dlech committed:
Update based on feedback

Technically COM != PInvoke, so describe the usage restrictions in a more generic way.

dlech commented:
Whoops, forgot about this one. I have added the change now.

yizhang82 reviewed (approved):
LGTM. Thank you!
